modules/nf-core/samshee/main.nf

```diff
@@ -0,0 +1,55 @@
+process SAMSHEE {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'docker://community.wave.seqera.io/library/pip_samshee:733e11f3377fc2e3' :
+        'community.wave.seqera.io/library/pip_samshee:733e11f3377fc2e3' }"
+
+    input:
+    tuple val(meta), path(samplesheet)
+    val(json_schema_validator)  // optional
+    val(name_schema_validator)  // optional
+    path(file_schema_validator) // optional
+
+    output:
+    tuple val(meta), path("*_formatted.csv"), emit: samplesheet
+    path "versions.yml"                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def arg_json_schema_validator = json_schema_validator ? "--schema '${json_schema_validator}'"                         : ""
+    def arg_name_schema_validator = name_schema_validator ? "--schema '${name_schema_validator}'"                         : ""
```

Comment:
I think we could either merge these two into one channel, then the user would need to specify `'{"$ref": "urn:samshee:illuminav2/v1"}` or I'd ajust the second line as follows: 

```suggestion
    def arg_json_schema_validator = json_schema_validator ? "--schema '${json_schema_validator}'"                         : ""
    def arg_name_schema_validator = name_schema_validator ? "--schema '{\"\$ref\": \"${name_schema_validator}\"'"                         : ""
```

I have a slight preference for the single channel version as it keeps the module simpler.

Reply:
@grst I also thought of having the user just specify just the name, but I thought it could add a layer of confusion. I did it this way so that they can just use exactly the same strings that are displayed in the samshee readme. Doing it this way we would need to add an explanation saying that they need to extract the string from the readme that comes after the "$ref:",  do you think it would be easier for them to understand?

Reply:
fair enough, let's go with a single channel for schemas specified as string then (+ the second one for `path`s).